modules/pricer/src/main/java/com/opengamma/strata/pricer/swaption/NormalSabrParametersSwaptionVolatilities.java

```diff
@@ -0,0 +1,1004 @@
+import java.util.Optional;
+
+import org.joda.beans.Bean;
+import org.joda.beans.ImmutableBean;
+import org.joda.beans.JodaBeanUtils;
+import org.joda.beans.MetaBean;
+import org.joda.beans.MetaProperty;
+import org.joda.beans.gen.BeanDefinition;
+import org.joda.beans.gen.PropertyDefinition;
+import org.joda.beans.impl.direct.DirectFieldsBeanBuilder;
+import org.joda.beans.impl.direct.DirectMetaBean;
+import org.joda.beans.impl.direct.DirectMetaProperty;
+import org.joda.beans.impl.direct.DirectMetaPropertyMap;
+
+import com.google.common.collect.ImmutableList;
+import com.opengamma.strata.basics.date.DayCount;
+import com.opengamma.strata.basics.value.ValueDerivatives;
+import com.opengamma.strata.collect.ArgChecker;
+import com.opengamma.strata.collect.array.DoubleArray;
+import com.opengamma.strata.data.MarketDataName;
+import com.opengamma.strata.market.model.SabrParameterType;
+import com.opengamma.strata.market.param.CurrencyParameterSensitivities;
+import com.opengamma.strata.market.param.CurrencyParameterSensitivity;
+import com.opengamma.strata.market.param.ParameterMetadata;
+import com.opengamma.strata.market.param.ParameterPerturbation;
+import com.opengamma.strata.market.param.UnitParameterSensitivity;
+import com.opengamma.strata.market.sensitivity.PointSensitivities;
+import com.opengamma.strata.market.sensitivity.PointSensitivity;
+import com.opengamma.strata.market.surface.Surface;
+import com.opengamma.strata.pricer.impl.option.NormalFormulaRepository;
+import com.opengamma.strata.pricer.model.SabrInterestRateParameters;
+import com.opengamma.strata.product.common.PutCall;
+import com.opengamma.strata.product.swap.type.FixedFloatSwapConvention;
+
+/**
+ * Volatility environment for swaptions in the SABR model.
+ * <p>
+ * The volatility is represented in terms of SABR model parameters.
+ * <p>
+ * The parameters shift surface is not used.
+ */
+@BeanDefinition
+public final class NormalSabrParametersSwaptionVolatilities
+    implements NormalSabrSwaptionVolatilities, ImmutableBean, Serializable {
+
+  /**
+   * The name.
+   */
+  @PropertyDefinition(validate = "notNull", overrideGet = true)
+  private final SwaptionVolatilitiesName name;
+  /**
+   * The swap convention that the volatilities are to be used for.
+   */
+  @PropertyDefinition(validate = "notNull", overrideGet = true)
+  private final FixedFloatSwapConvention convention;
+  /**
+   * The valuation date-time.
+   * <p>
+   * The volatilities are calibrated for this date-time.
+   */
+  @PropertyDefinition(validate = "notNull", overrideGet = true)
+  private final ZonedDateTime valuationDateTime;
+  /**
+   * The SABR model parameters.
+   * <p>
+   * Each model parameter of SABR model is a surface.
+   * The x-value of the surface is the expiry, as a year fraction.
+   * The y-value of the surface is the swap tenor, as a year fraction rounded to the month.
+   */
+  @PropertyDefinition(validate = "notNull")
+  private final SabrInterestRateParameters parameters;
```

Comment:
`ValueType` for `NormalSabrSwaptionVolatilities#getVolatilityType` does not necessarily match that for `SabrVolatilityFormula#getVolatilityType` in `SabrInterestRateParameters`. Do we need to add ArgChecker or is this intended?

Reply:
The check is indeed missing. Adding it and updating the tests.